src/jog/dto/create-jog.dto.ts

```diff
@@ -0,0 +1,35 @@
+import {IsDate, IsEmpty, IsNotEmpty, IsNumber, IsObject} from "class-validator";
+import {Type} from "class-transformer";
+import {User} from "../../auth/schemas/user.schema";
+
+class Location{
+    @IsNotEmpty()
+    @IsNumber()
+    lat: number;
+
+    @IsNotEmpty()
+    @IsNumber()
+    lon: number;
+}
+
+export class CreateJogDto{
+    @IsNotEmpty()
+    @IsNumber()
+    readonly time: number;
+
+    @IsNotEmpty()
+    @IsNumber()
+    readonly distance: number;
+
+    @IsNotEmpty()
+    @IsDate()
+    readonly date: Date;
+
+    @IsNotEmpty()
+    @IsObject()
+    @Type(() => Location)
+    readonly location: Location;
+
+    @IsEmpty({message: 'User is not allowed to be set'})
+    readonly user: User;

```

Comment:
Passing a user as a body request is not a good idea. That means a user can create a `jog` for another user. The user should be got from JWT token.

Reply:
but doesn't the @IsEmpty decorator ensure that it can't be passed by a user?